### DISCUSSION TITLE
Is there a way to prevent layers from being renameable by the user?

### DISCUSSION THREAD

**Discussion by glaprida** [GrapesJS/grapesjs#5538] [Q&A]
I am dynamically setting the layer names for text component so that it shows the first X characters of its content, but I don't want users to be abe to rename them (or any other layer)

**Comment by mosh-tudor** [comment]
@glaprida 

I don't see such functionality in https://github.com/GrapesJS/grapesjs/blob/dev/src/navigator/view/ItemView.ts

But there is the dirty method -> using `pointer-events: none;`

**Reply by glaprida** [reply]
I don't think that will work for me... I still need to be able to select the layers.

I see on the link you shared that there's an event `'dblclick [data-name]': 'handleEdit',`

maybe there's some way to unsubscribe from that event; or maybe listen to it from a global hook (some thing like `editor.on('layer:edit')` ) and return `false`

**Reply by artf** [reply]
Well, it should actually work if you apply `pointer-events: none;` to `.gjs-layer-name` 😅
